src/nouislider.js

```diff
@@ -408,7 +408,7 @@
     Spectrum.prototype.getMargin = function(value) {
         var step = this.xNumSteps[0];
 
-        if (step && (value / step) % 1 !== 0) {
+        if (step && Math.round(value / step) % 1 !== 0) {
```

Comment:
I see why you are making this change, but I'm afraid this won't work with slider with non-integer ranges/steps.